File: javaee/persistence/eclipselink/src/main/java/org/sgodden/tom/domain/CustomerOrderRepository.java
package org.sgodden.tom.domain;

import com.google.inject.ImplementedBy;

/**
 * Repository for the {@link CustomerOrder} entity.
 * @author sgodden
 */
@ImplementedBy(CustomerOrderRepositoryImpl.class)
public interface CustomerOrderRepository {
	
	void persist(CustomerOrder order);
	
	/**
	 * Returns the {@link CustomerOrder} instance
	 * having the passed id.
	 * @param id the id.
	 * @return the customer order instance, or <code>null</code> if one does not exist.
	 */
	CustomerOrder findById(long id);
	
	/**
	 * Returns the number of customer orders.
	 * @return the number of customer orders.
	 */
	long count();

}

File: templates/scala-mongodb/model-api/src/main/scala/org/sgodden/tom/model/CustomerOrderRepository.scala
package org.sgodden.tom.model

trait CustomerOrderRepository {

  def remove(order: ICustomerOrder)

  def findAll: List[ICustomerOrder]

  def persist(order: ICustomerOrder)

  def merge(order: ICustomerOrder)

  def findById(id: String): ICustomerOrder

  def count: Long
}
